qutip/qip/circuit.py

```diff
@@ -593,47 +592,36 @@ def add_circuit(self, qc, start=0):
-                                 "SNOT", "SQRTNOT", "PHASEGATE", "QASMU"]:
-                    self.add_gate(gate.name, gate.targets[0] + start, None,
-                                  gate.arg_value, gate.arg_label)
-                elif gate.name in ["X", "Y", "Z", "S", "T"]:
-                    self.add_gate(gate.name, gate.targets[0] + start, None,
-                                  None, gate.arg_label)
-                elif gate.name in ["CPHASE", "CNOT", "CSIGN", "CRX", "CRY",
-                                   "CRZ", "CY", "CZ", "CS", "CT"]:
-                    self.add_gate(gate.name, gate.targets[0] + start,
-                                  gate.controls[0] + start, gate.arg_value,
-                                  gate.arg_label)
-                elif gate.name in ["BERKELEY", "SWAPalpha", "SWAP", "ISWAP",
-                                   "SQRTSWAP", "SQRTISWAP"]:
-                    self.add_gate(gate.name,
-                                  [gate.targets[0] + start,
-                                   gate.targets[1] + start])
-                elif gate.name in ["TOFFOLI"]:
-                    self.add_gate(gate.name, gate.targets[0] + start,
-                                  [gate.controls[0] + start,
-                                   gate.controls[1] + start], None, None)
-                elif gate.name in ["FREDKIN"]:
-                    self.add_gate(gate.name,
-                                  [gate.targets[0] + start,
-                                   gate.targets[1] + start],
-                                  gate.controls + start, None, None)
-                elif gate.name in self.user_gates:
-                    self.add_gate(
-                        gate.name, targets=gate.targets,
-                        arg_value=gate.arg_value)
-            else:
-                measurement = circuit_op
+                # gate = circuit_op
+
+                if circuit_op.targets is not None:
+                    tar = [target + start for target in circuit_op.targets]
+                else:
+                    tar = None
+                if circuit_op.controls is not None:
+                    ctrl = [control + start for control in circuit_op.controls]
+                else:
+                    ctrl = None
+
+                self.add_gate(
+                    circuit_op.name, targets=tar, controls=ctrl,
+                    arg_value=circuit_op.arg_value)
+            elif isinstance(circuit_op, Measurement):
                 self.add_measurement(
-                                measurement.name,
-                                targets=[measurement.targets[0] + start],
-                                classical_store=measurement.classical_store)
+                                circuit_op.name,
+                                targets=[target + start for target in circuit_op.targets],
+                                classical_store=circuit_op.classical_store)
```

Comment:
We can remove some indents, 16 spaces are unnecessary and it also makes code climate complain.

Reply:
Sure. I've shortened the lines